### DISCUSSION TITLE
Improve `{% js %}` ability to register a CDN URL (ie: Alpine.js)

### DISCUSSION THREAD

**Discussion by lindseydiloreto** [craftcms/cms#10666] [Ideas] [answered]
### ✅ Situation

The `js` tag can register a JavaScript file as described [here...](https://craftcms.com/docs/3.x/dev/tags.html#js)

```twig
{% js "/assets/js/script.js" %}
```

### ⚠️ Problem

However, there is a caveat...

<img width="453" alt="To register a JavaScript file, the URL must end in .js" src="https://user-images.githubusercontent.com/5309692/156470137-2c8a3556-b66a-4fe2-bf4f-feeb5633f22e.png">

While this works just fine in 90% of cases, there are valid use cases in which this rule is too strict. A prime example (and the reason I am here today) is the CDN for [Alpine.js](https://alpinejs.dev):

```js
<script src="//unpkg.com/alpinejs" defer></script>
```

Lacking a trailing `.js`, the Alpine CDN can't be registered with the `js` tag. 😢

```twig
{# Neither of these work #}
{% js "//unpkg.com/alpinejs" %}
{% js "https://unpkg.com/alpinejs" %}
```

### 💡 Suggestion

Can we add a few other identifiers for a potential file match?

 - If the URL begins with `http(s)://`
 - If the URL begins with `//`

That second one might be a little tricky (because it could be a JS comment), but the first one should be pretty straightforward.

This would scoop up most CDN URLs, and bring that 90% coverage up to 99% (roughly approximated without actual statistics).

Let me know your thoughts. Thanks!

**Comment by brandonkelly** [comment, marked as answer]
Just released 3.7.36 with a new check to start treating any URLs that begin with `https://` or `http://` as external JS/CSS URLs.

**Reply by lindseydiloreto** [reply]
Woohoo, that's perfect! Thanks @brandonkelly! 🍺

**Reply by proimage** [reply]
Just in case it helps anyone who for some reason can't update a Craft install to that version, the workaround I've used is to add a parameter to the URL that satisfies the `.js` requirement, eg:
`{% js "https://unpkg.com/alpinejs?ext=.js" %}`
